### PR TITLE
Remove ContentTag, add A::rel() method, fix tests

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: tests/Tag/ATest.php [hunk 1]
```diff
@@ -65,6 +65,25 @@ public function testMailto(string $expected, ?string $url): void
...
+        return [
+            ['<a></a>', null],
+            ['<a rel="nofollow"></a>', 'nofollow'],
+            ['<a rel="noopener"></a>', 'noopener'],
+            ['<a rel="noreferrer"></a>', 'noreferrer'],
+            ['<a rel="nofollow noopener noreferrer"></a>', 'nofollow noopener noreferrer'],
+        ];
+    }
+
+    /**
```
Comment (vjik) [inline tests/Tag/ATest.php:79]:
```suggestion
      /**
     * @dataProvider dataRel
     */
    public function testRel(string $expected, ?string $rel): void
    {
        $this->assertSame($expected, (string)A::tag()->rel($rel));
    }
```

File: tests/Tag/Base/ContentTagTest.php [hunk 2]
```diff
@@ -9,32 +9,32 @@
 use Yiisoft\Html\Tag\P;
 use Yiisoft\Html\Tag\Span;
 use Yiisoft\Html\Tests\Objects\StringableObject;
-use Yiisoft\Html\Tests\Objects\TestContentTag;
+use Yiisoft\Html\Tests\Objects\TestTagContentTrait;
 use function is_array;
 
 final class ContentTagTest extends TestCase
```
Comment (vjik) [inline tests/Tag/Base/ContentTagTest.php:15]: Better rename to `TagContentTraitTest`

File: tests/Tag/ATest.php [hunk 3]
```diff
@@ -78,15 +97,16 @@ public function dataTarget(): array
...
         $tag = A::tag();
-        self::assertNotSame($tag, $tag->href(null));
-        self::assertNotSame($tag, $tag->url(null));
-        self::assertNotSame($tag, $tag->mailto(null));
-        self::assertNotSame($tag, $tag->target(null));
+        $this->assertNotSame($tag, $tag->content(''));
+        $this->assertNotSame($tag, $tag->href(null));
+        $this->assertNotSame($tag, $tag->url(null));
+        $this->assertNotSame($tag, $tag->mailto(null));
+        $this->assertNotSame($tag, $tag->target(null));
```
Comment (vjik) [inline tests/Tag/ATest.php:110]: Need test of method `rel()` on immutability.

Comment (devanych) [reply tests/Tag/ATest.php:111]: Done.

File: tests/Tag/Base/ContentTagTest.php (hunk 2 above)
Comment (devanych) [reply tests/Tag/Base/ContentTagTest.php]: Done

File: tests/Tag/ATest.php (hunk 1 above)
Comment (devanych) [reply tests/Tag/ATest.php:79]: Done

**Review by vjik — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**